frontend/simple/views/components/Contribution/Contribution.vue

```diff
@@ -0,0 +1,171 @@
+      @keyup.enter="handleSubmit"
+    >
+
+    <i18n tag="button" class="button" @click="cancel">Cancel</i18n>
+    <i18n tag="button" class="button has-text-primary" v-if="isAdding && isFilled" @click="handleSubmit">Add</i18n>
+    <i18n tag="button" class="button has-text-primary" v-if="isEditing && isFilled" @click="handleSubmit">Save</i18n>
+    <i18n tag="button" class="button has-text-danger" v-if="isEditing && !isFilled" @click="handleSubmit">Delete</i18n>
+  </li>
+
+  <li v-else-if="isEditable" :class="itemClasses">
+    <p class="box-body">
+      <slot></slot>
+    </p>
+    <div class="box-controls">
+      <button
+        class="button is-icon"
+        :aria-label="editAriaLabel"
+        @click="handleEditClick"
+      >
+        <i class="fa" :class="iconClass" aria-hidden="true"></i>
+      </button>
+    </div>
+  </li>
+
+  <li v-else-if="isUnfilled">
+    <button :class="itemClasses" @click="handleClick">
+      <slot></slot>
+    </button>
+  </li>
+
+  <li v-else :class="itemClasses">
+    <slot></slot>
+  </li>
+</template>
+<style lang="scss" scoped>
+@import "../../../assets/sass/theme/index";
+
+.c-contribution,
+.c-form {
+  min-height: 45px;
+}
+
+.c-contribution {
+  width: 100%;
+  font-size: inherit;
+
+  &.is-default {
+    border: none;
+    background-color: $white-ter;
+  }
+
+  &.is-editable {
+    border: none;
+    background-color: $primary-bg-a;
+  }
+
+  &.is-unfilled {
+    cursor: pointer;
+    text-align: left;
+
+    &:hover,
+    &:focus {
+      background-color: $primary-bg-a;
+    }
+  }
+
+  &-icon {
+    color: $primary;
+    margin-right: $gi-spacer-sm;
+  }
+}
+</style>
+<script>
+export default {
+  name: 'Contribution',
+  props: {
+    variant: {
+      type: String,
+      validator (value) {
+        return [
+          'default', // grey box just to read the content
+          'unfilled', // dashed box interactive to add a new contribution
+          'editable' // blue box to edit a contribution
+        ].indexOf(value) > -1
+      },
+      default: 'default'
+    }
+  },
+  data () {
+    return {
+      isAdding: false,
+      isEditing: false,
+      isFilled: null, // when true, show add/save button.
+      placeholders: [this.L('Portuguese classes'), this.L('Programming'), this.L('Cooking'), this.L('Parties'), this.L('Free cinema tickets')]
+    }
+  },
+  computed: {
+    itemClasses () {
+      return [
+        'box is-compact c-contribution',
+        `is-${this.variant}`,
+        { 'has-controls': this.isEditable }
+      ]
+    },
+    isEditable () {
+      return this.variant === 'editable'
+    },
+    isUnfilled () {
+      return this.variant === 'unfilled'
+    },
+    editAriaLabel () {
+      return this.L('Edit contribution settings')
+    },
+    randomPlaceholder () {
+      return this.placeholders[Math.floor(Math.random() * this.placeholders.length)]
+    },
+    iconClass () {
+      return this.$listeners.click ? 'fa-ellipsis-v' : 'fa-edit'
```

Comment:
(Read this comment after the one below about `this.$listeners`)

Similarly, here I would rely on some internal `this.state` variable that has a value of possible enums (defined as constants at the top of this file, perhaps)

Reply:
possible enums? like what? check what I've done to see if it's ok...